benchmarks/src/commonMain/kotlin/BufferOps.kt

```diff
@@ -384,3 +385,33 @@ open class BufferReadNewByteArray : BufferRWBenchmarkBase() {
         return buffer.readByteArray(size)
     }
 }
+
+@State(Scope.Benchmark)
+open class IndexOfByteString {
+    @Param("1024:2", "8192:2", "10000:2", "10000:8")
+    var params: String = "<buffer size>:<bytestring size>"
+
+    private var buffer = Buffer()
+    private var byteString = ByteString()
+
+    @Setup
+    fun setup() {
+        val paramsParsed = params.split(':').map { it.toInt() }.toIntArray()
+        require(paramsParsed.size == 2)
+
+        val bufferSize = paramsParsed[0]
+        val bsSize = paramsParsed[1]
+        byteString = ByteString(ByteArray(bsSize) { 0x42 })
+
+        for (idx in 0 until bufferSize) {
+            if (idx % bsSize == 0) {
+                buffer.writeByte(0)
+            } else {
+                buffer.writeByte(0x42)
+            }
+        }
+    }
+
+    @Benchmark
+    fun benchmark() = buffer.indexOf(byteString)
```

Comment:
In this case, the result of `indexOf` will always be 0?

Reply:
In fact, every byte at a position whose index is a multiple of bytestring size is `0`, so the benchmark has to scan the whole buffer, every time looking for up to `length-1` bytes ahead.